### PR TITLE
Onnx: fix squeeze and dropout

### PR DESCRIPTION
**Pull request by ym1234**
OnnxBackendPyTorchConvertedModelTest eats up too much ram triggering my oom, so these results are without it.
 
`242 failed, 500 passed, 1892 skipped, 2 warnings in 25.76s`

### CODE REVIEW

File: extra/onnx.py [hunk 1]
```diff
@@ -118,7 +117,9 @@ def run_onnx(inputs={}, debug=False):
       elif n.op_type == "Elu": ret = inp[0].elu(alpha=opt.get('alpha', 1.0))
       elif n.op_type == "Concat": ret = inp[0].cat(*inp[1:], dim=opt['axis'])
       elif n.op_type == "Transpose": ret = inp[0].permute(order=opt.get('perm', list(range(len(inp[0].shape))[::-1])))
-      elif n.op_type == "Squeeze": ret = inp[0].reshape([s for i,s in enumerate(inp[0].shape) if i not in opt['axes']])
+      elif n.op_type == "Squeeze":
+        axes = (inp[1].numpy() + len(inp[0].shape)) % len(inp[0].shape)
```
Comment (dc-dc-dc) [inline extra/onnx.py:121]: use safe_numpy here as it will use a cache if exists

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/tensor.py                         2               2
------------------------------------------------------------
lines added in the tinygrad folder: 0
```

**Comment by ym1234** [comment]
`198 failed, 544 passed, 1892 skipped, 2 warnings in 26.12s` after rebasing + fixing range and reduce

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ def rand(*shape, **kwargs):
   # ***** creation helper functions *****
 
   @staticmethod
-  def full(shape:Tuple[int, ...], fill_value, **kwargs): return Tensor(fill_value, **kwargs).reshape([1]*len(new_shape := argfix(shape))).expand(new_shape).contiguous()
+  def full(shape:Tuple[int, ...], fill_value, **kwargs): return Tensor(fill_value, **kwargs).reshape([1]*len(new_shape := tuple(map(int, argfix(shape))))).expand(new_shape).contiguous()
```
Comment (geohot) [inline tinygrad/tensor.py:137]: Why would the input here not be an int?

Comment (ym1234) [reply tinygrad/tensor.py]: Yeah it should be fixed in arange

**Comment by ym1234** [comment]
... was running the wrong efficientnet test, should be fixed now.

File: extra/onnx.py [hunk 3]
```diff
@@ -99,8 +99,6 @@ def run_onnx(inputs={}, debug=False):
         else:
           input_tensors[inp.name] = Tensor(inputs[inp.name], requires_grad=False)
         input_shape = input_tensors[inp.name].shape
-        if input_shape == (0,): raise NotImplementedError("empty tensors aren't supported in tinygrad")
-        assert input_shape == shape, f"wrong shape for input {inp.name}, {input_shape} isn't {shape}"
```
[the remark this answers is not in the document]
Comment (geohot) [reply extra/onnx.py:103]: Yea, waiting on response for this. 102 can be removed, but not 103

**Comment by geohot** [comment]
Closing, need a response to the 103 line removal. You are welcome to reopen.
